src/js/jsx/sections/pages/LayerFace.jsx

```diff
@@ -40,16 +40,48 @@ define(function (require, exports, module) {
         system = require("js/util/system"),
         strings = require("i18n!nls/strings");
 
+    /*
+     * Function for checking whether React component should update
+     * Passed to Droppable composed component in order to save on extraneous renders
+     *
+     * @param {Object} nextProps - Next set of properties for this component
+     * @return {boolean}
+     */
+    var shouldComponentUpdate = function (nextProps) {
+        // Drag states
+        if (this.props.dragTarget !== nextProps.dragTarget ||
+            this.props.dropAbove !== nextProps.dropAbove ||
+            this.props.dragPosition !== nextProps.dragPosition ||
+            this.props.dragStyle !== nextProps.dragStyle ||
+            this.props.dropTarget !== nextProps.dropTarget) {
+            return true;
+        }
+        
+        // Face change
+        if (!Immutable.is(this.props.layer.face, nextProps.layer.face)) {
+            return true;
+        }
+                      
+        // Deeper selection changes
+        var document = this.props.document,
+            allSelected = document.layers.allSelected,
+            childOfSelection = allSelected.includes(this.props.layer),
+            parentSelectedChanged = false,
+            nextAllSelected = nextProps.document.layers.allSelected;
+
+        if (childOfSelection) {
+            parentSelectedChanged = !Immutable.is(document.layers.selected, nextProps.document.layers.selected);
+        }
+
+        return allSelected.includes(this.props.layer) !== nextAllSelected.includes(nextProps.layer) ||
```

Comment:
Reverse the disjuncts to save some executions of the slightly more expensive-to-compute one.